src/filesbutton.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * @copyright Copyright (c) 2020 cnmicha <cnmicha@bhb-networks.com>
+ *
+ * @author cnmicha <cnmicha@bhb-networks.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import jQuery from 'jquery'
+import { generateUrl } from '@nextcloud/router'
+
+/**
+ * Adds a button to the files app that opens the current directory in the photos app.
+ */
+(function() {
+	if (jQuery('#filesApp').val()) {
+		const button = jQuery(`
+			<label id="open-in-photos-button" class="button icon-picture" title="Show in photos"></label>
+		`)
+
+		button.click(function() {
+			replaceButtonWithLoadingSpinner(this)
+			redirectToPhotosApp()
+		})
+
+		// Add button to UI
+		jQuery('#showgridview').after(button)
+	}
+})()
+
+function replaceButtonWithLoadingSpinner(button) {
+	jQuery(button).removeClass('button').removeClass('icon-picture').addClass('icon-loading-small')
+}
+
+function redirectToPhotosApp() {
+	window.location.href = buildPhotosUrl()
```

Comment:
You can also directly use a link instead of a button. That would be more adequate regarding the html semantics

Reply:
I agree, but I don't see how this is possible.

In https://github.com/cnmicha/photos/blob/feature/upstream-566/src/FilesButton.vue#L50-L52, the `buildPhotosUrl()` function seems to only work when the files app has done some internal logic. Before (and on vue init time), `FileList.getCurrentDirectory()` is not available.

Reply:
Use an ActionLink instead of an ActionButton
https://nextcloud-vue-components.netlify.com/

Reply:
@skjnldsv Done but the pipeline seems to be stuck (the checks don't report results). I already tried again by changing the commit id and force-pushing, but no luck

Reply:
@skjnldsv Any news on this?